packages/frontend-2/components/workspace/header/Header.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <div v-if="workspaceInfo" class="flex justify-between items-center">
+    <div class="flex gap-2 mb-3 mt-2">
+      <Component
+        :is="workspaceInfo.logo"
+        v-if="workspaceInfo.logo"
+        class="w-5 h-5 mt-0.5"
+      />
+      <div class="flex flex-col">
+        <h1 class="text-heading-lg">{{ workspaceInfo.name }}</h1>
+        <div class="text-body-xs text-foreground-2">
+          {{ workspaceInfo.description || 'No workspace description' }}
+        </div>
+      </div>
+    </div>
+    <div class="flex items-center gap-2">
+      <div
+        class="text-body-3xs bg-foundation-2 text-foreground-2 rounded px-3 py-1 font-medium select-none"
+      >
+        {{ workspaceInfo.projects.totalCount || 0 }} Project{{
+          workspaceInfo.projects.totalCount === 1 ? '' : 's'
+        }}
+      </div>
+      <UserAvatarGroup
+        :users="workspaceInfo.team.map((teamMember) => teamMember.user)"
+        class="max-w-[104px]"
+      />
+      <FormButton color="outline">Invite</FormButton>
+      <WorkspaceHeaderActions />
+    </div>
+  </div>
+  <div v-else-if="loading">Loading...</div>
+  <div v-else-if="error">Error loading workspace data.</div>
+</template>
+
+<script setup lang="ts">
+import { useQuery } from '@vue/apollo-composable'
+import { workspacePageQuery } from '~~/lib/workspaces/graphql/queries'
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const {
+  result: workspaceInfoResult,
+  loading,
+  error
+} = useQuery(workspacePageQuery, () => ({
```

Comment:
Why is a header component invoking the query for the entire page? The query should be invoked at the top level where its first needed, and then fed down into child components through fragments. I can see that this same query is also independently invoked in WorkspaceProjectList - that's probably the best place for it (as nothing above it needs any of that data, but its own child components do).